agent/src/beerocks/slave/1905_beacon_query_to_vs.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _1905_BEACONB_QUERY_TO_VS_H_
+#define _1905_BEACONB_QUERY_TO_VS_H_
+
+#include <beerocks/tlvf/beerocks_message_monitor.h>
+#include <tlvf/CmduMessageRx.h>
+#include <tlvf/wfa_map/tlvBeaconMetricsQuery.h>
+
+namespace gate {
+
+// loads lhs with rhs. as if it was written: lhs = rhs; or lhs << rhs;
+// however load() was chosen over operator=() or operator<<() as it permits returning success/fail with the bool return value,
```

Comment:
1. from the code: `and also it opens the possibilities for adding additional parameters in the future if needed`
2. operator in this case is non natural:
user code: 
```cpp
auto vsPtr = message_com::create_vs_message</*....*/>( /*....*/);
vsPtr = rx; // operator=() may change lhs to point to nullptr, this is strange. normally you don't expect this. Especially you don't expect that operator=() will take care for life cycle of the given pointer. In this case it is might be deleting it in case the operation fails.
if (vsPtr) // ??? maybe, I don't know.
```
```cpp
vsPtr << rx // even less natural. you wouldn't imagine that vsPtr will point to something else at the end of this operation.
```
Moreover, I really wanted to write the load function to have `beerocks_message::cACTION_MONITOR_CLIENT_BEACON_11K_REQUEST &` as its input parameter, I don't know why I finally decided not to.